### PR TITLE
CAMEL-17945: Fix the unstable test HazelcastSedaTransferExchangeTest

### PR DESCRIPTION
**Pull request by essobedo**
Fix for https://issues.apache.org/jira/browse/CAMEL-17945

## Motivation

While working on [CAMEL-17921](https://issues.apache.org/jira/browse/CAMEL-17921), I noticed that the test `HazelcastSedaTransferExchangeTest` was unstable and should be fixed.

## Modifications

* Rewrite the test `testExchangeTransferDisabled` to use the new method `expectedNoHeaderReceived` instead of asserting for an `AssertionError`
* Cleanup the test class by renaming properly the methods annotated with `@BeforeAll` and `@AfterAll` (not related to the issue)
* Cleanup the test class by resetting the mock endpoint in a method annotated with `@AfterEach` (not related to the issue)
* Cleanup the test class by removing exception in the throw declaration when not needed (not related to the issue)

### CODE REVIEW

File: .github/workflows/master-pr-build.yml [hunk 1]
```diff
@@ -71,14 +71,14 @@ jobs:
...
-          name: build.log
-          path: build.log
+#      - name: maven build
+#        run: ${{ steps.install-mvnd.outputs.mvnd-dir }}/mvnd -l build.log -Dmvnd.threads=2 -V -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false -Dmaven.wagon.httpconnectionManager.ttlSeconds=120 --no-transfer-progress -Pfastinstall -Dcheckstyle.skip=true -DskipTests install
+#      - name: archive logs
+#        uses: actions/upload-artifact@v3
+#        if: always()
+#        with:
+#          name: build.log
+#          path: build.log
```
Comment (orpiske) [inline .github/workflows/master-pr-build.yml:81]: This is preventing the build logs from being uploaded.

**Review by orpiske — CHANGES_REQUESTED** [review]
Added a comment about the build log.

File: .github/workflows/master-pr-build.yml (hunk 1 above)
Comment (essobedo) [reply .github/workflows/master-pr-build.yml:81]: I know, it is only to get a quick feedback (avoid waiting for the maven build) as the commit message states. I will of course remove this commit once the quick test will pass, sorry for the noise

Comment (orpiske) [reply .github/workflows/master-pr-build.yml:81]: No worries, thanks for the heads up.

**Review by orpiske — APPROVED** [review]

**Comment by essobedo** [comment]
Waiting for https://github.com/apache/camel/pull/7420

**Review by davsclaus — APPROVED** [review]

**Pull request merged**